ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutor.java

```diff
@@ -229,6 +241,22 @@ public void buildAndExecuteQuery(
     }
   }
 
+  private static JsonNode fromSchemaMetadata(final Optional<SchemaMetadata> metadata) {
+    if (!metadata.isPresent()) {
+      return NullNode.getInstance();
+    }
+
+    if (ProtobufFormat.NAME.equals(metadata.get().getSchemaType())) {
```

Comment:
Can we add a comment explaining that we need this special case because protobuf schemas aren't valid JSON? If I hadn't coincidentally encountered this in SerdeUtil ([link](https://github.com/confluentinc/ksql/blob/807549c1b979d393137244c2586c575f59d6df1a/ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/utils/SerdeUtil.java#L100)) the other day, I would've been confused by this.

Relatedly, it sucks that we have to duplicate this special-casing in multiple places... can't think of a good alternative at the moment though.

Reply:
I was originally going to refactor all of it to just be a string, but then that made the QTT tests really annoying to write (escaped JSON inside JSON strings really sucks)